src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -70,6 +71,7 @@ public class KestrelServerOptions
         /// <remarks>
         /// Defaults to true.
         /// </remarks>
+        [Obsolete("This property is obsolete and will be removed in a future version. Use ListenOptions.NoDelay instead.")]
         public bool NoDelay { get; set; } = true;
```

Comment:
You're already making a bunch of breaking changes, why obsolete this instead of removing it?